packages/codegen-ui-react/lib/__tests__/utils/__snapshots__/format.test.ts.snap

```diff
@@ -0,0 +1,103 @@
+export function formatDate(date: string, format: DateFormat[\\"dateFormat\\"]): string {
+    if (date === undefined || date === null) {
+        return date;
+    }
+    const validDate = new Date(Date.parse(date));
+    if (validDate.toString() === invalidDateStr) {
+        return date;
+    }
+    const splitDate = date.split(/-|+|Z/);
+    const year = splitDate[0];
+    const month = splitDate[1];
+    const day = splitDate[2];
+    const truncatedMonth = month.replace(/^0+/, \\"\\");
+    switch (format) {
+        case \\"locale\\": return validDate.toLocaleDateString();
+        case \\"YYYY.MM.DD\\": return \`\${year}.\${month}.\${day}\`;
+        case \\"DD.MM.YYYY\\": return \`\${day}.\${month}.\${year}\`;
+        case \\"MM/DD/YYYY\\": return \`\${month}/\${day}/\${year}\`;
+        case \\"Mmm DD, YYYY\\": return \`\${monthToShortMon[truncatedMonth]} \${day}, \${year}\`;
+        default: return date;
+    }
+}
+export function formatTime(time: string, format: TimeFormat[\\"timeFormat\\"]): string {
+    if (time === undefined || time === null) {
+        return time;
+    }
+    const splitTime = time.split(/:|Z/);
+    if (splitTime.length < 3) {
+        return time;
+    }
+    const validTime = new Date();
+    validTime.setHours(Number.parseInt(splitTime[0], 10));
+    validTime.setMinutes(Number.parseInt(splitTime[1], 10));
+    const splitSeconds = splitTime[2].split(\\".\\");
+    validTime.setSeconds(Number.parseInt(splitSeconds[0], 10), Number.parseInt(splitSeconds[1], 10));
+    if (validTime.toString() === invalidDateStr) {
+        return time;
+    }
+    switch (format) {
+        case \\"locale\\": return validTime.toLocaleTimeString();
+        case \\"hours24\\": return validTime.toLocaleTimeString(\\"en-gb\\");
+        case \\"hours12\\": return validTime.toLocaleTimeString(\\"en-us\\");
+        default: return time;
+    }
+}
+export function formatDateTime(dateTimeStr: string, format: DateTimeFormat[\\"dateTimeFormat\\"]): string {
+    if (dateTimeStr === undefined || dateTimeStr === null) {
+        return dateTimeStr;
+    }
+    const dateTime = /^d+$/.test(dateTimeStr) ? new Date(Number.parseInt(dateTimeStr, 10)) : new Date(Date.parse(dateTimeStr));
+    if (dateTime.toString() === invalidDateStr) {
+        return dateTimeStr;
+    }
+    if (format === \\"locale\\") {
+        return dateTime.toLocaleString();
+    }
+    const dateAndTime = dateTime.toISOString().split(\\"T\\");
+    const date = formatDate(dateAndTime[0], format.dateFormat);
+    const time = formatTime(dateAndTime[1], format.timeFormat);
+    return \`\${date} - \${time}\`;
+}
+export function format(value: string, formatterInput: FormatInputType) {
+    switch (formatterInput.type) {
+        case \\"DateFormat\\": return formatDate(value, formatterInput.format);
+        case \\"DateTimeFormat\\": return formatDateTime(value, formatterInput.format);
+        case \\"TimeFormat\\": return formatTime(value, formatterInput.format);
```

Comment:
We want to handle undefined for `value` and `formatterInput`. Both should return the value as is

Reply:
we don't check for undefined before calling the function? Usually I check the inputs before.

Reply:
Yeah but we can't assume anything since it's codegenned for customers and they can provide their own input

Reply:
oh it's already being handled in the functions being called, formatDate, formatDateTime etc

Reply:
`formatterInput` can be undefined so that has to be handled before the switch statement otherwise it would cause an error. undefined values are passed in formatterInput when customers don't have their custom `format` map and they did not configure any formatting in UI Builder

Reply:
In that case there should be a default formatter. dates for example should always use a default format

Reply:
if we're passing undefined formatting configs, then I think there's something wrong with our code

Reply:
[This](https://github.com/aws-amplify/amplify-codegen-ui/pull/579/files#diff-13b04baba7c792bc0221d51e887f6ea6ecf8eb0349078e529b62cf06af628435R34) shouldn't blindly be calling formatter when we don't need it. Is it possible to make it smarter? If it's undefined, don't call formatter and just use the value.

Reply:
Yeah that's what I originally suggested. I don't mind going that route